src/goImport.ts

```diff
@@ -9,31 +9,95 @@ import vscode = require('vscode');
+		return Promise.all<string[]>([goPkgsPromise, importsPromise]).then(values => {
+			let pkgs = values[0];
+			let importedPkgs = values [1];
+
+			if (!vendorSupport || !vscode.window.activeTextEditor) {
+				if (importedPkgs.length > 0) {
+					pkgs = pkgs.filter(element => {
+						return importedPkgs.indexOf(element) === -1;
+					});
+				}
+				return pkgs.sort().slice(1);
+			}
+
+			let currentFileDirPath = path.dirname(vscode.window.activeTextEditor.document.fileName);
+			let workspaces: string[] = process.env['GOPATH'].split(path.delimiter);
+			let currentWorkspace = path.join(workspaces[0], 'src');
+
+			// Workaround for issue in https://github.com/Microsoft/vscode/issues/9448#issuecomment-244804026
+			if (process.platform === 'win32') {
+				currentFileDirPath = currentFileDirPath.substr(0, 1).toUpperCase() + currentFileDirPath.substr(1);
+			}
+
+			// In case of multiple workspaces, find current workspace by checking if current file is
+			// under any of the workspaces in $GOPATH
+			for (let i = 1; i < workspaces.length; i++) {
+				let possibleCurrentWorkspace = path.join(workspaces[i], 'src');
+				if (currentFileDirPath.startsWith(possibleCurrentWorkspace)) {
+					// In case of nested workspaces, (example: both /Users/me and /Users/me/src/a/b/c are in $GOPATH)
+					// both parent & child workspace in the nested workspaces pair can make it inside the above if block
+					// Therefore, the below check will take longer (more specific to current file) of the two
+					if (possibleCurrentWorkspace.length > currentWorkspace.length) {
+						currentWorkspace = possibleCurrentWorkspace;
+					}
+				}
+			}
+
+			let updatedPackages: string[] = [];
+			pkgs.forEach(pkg => {
+				if (!pkg || importedPkgs.indexOf(pkg) > -1) {
+					return;
+				}
+
+				let vendorIndex = pkg.indexOf('/vendor/');
+
+				// Check if current file and the vendor pkg belong to the same root project
+				// If yes, then vendor pkg can be replaced with its relative path to the "vendor" folder
+				if (vendorIndex > 0) {
+					let rootProjectForVendorPkg = path.join(currentWorkspace, pkg.substr(0, vendorIndex));
+					let relativePathForVendorPkg = pkg.substring(vendorIndex + 8);
+
+					if (relativePathForVendorPkg && currentFileDirPath.startsWith(rootProjectForVendorPkg)) {
+						// When a vendor package for a project/repo already exists outside the current project,
+						// pkgs list would already have an entry for it.
+						// In which case, we can skip adding the vendor package to the list
+						// Example: Assume package 'a' is installed from github.com/somerepo/a to $GOPATH
+						//          The current project github.com/someotherrepo/b may have a version of a as a vendor package
+						//          github.com/someotherrepo/b/vendor/github.com/somerepo/a
+						//          In this case, github.com/somerepo/a would already be returned by goPkgs
+						if (pkgs.indexOf(relativePathForVendorPkg) === -1) {
```

Comment:
set is good idea, though in the end I'll have to convert it to array to have sorting